evaluator/evaluator_mock.go

```diff
@@ -0,0 +1,65 @@
+// Code generated by mockery v2.10.0. DO NOT EDIT.
```

Comment:
I've used [https://github.com/vektra/mockery](mockery) to generate this mock, it's a very handy tool.
I'll open a task to generate mocks for all files as part of the CI process/ pre-commit.